iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -241,8 +240,9 @@ - (void)presentInAppMessage:(OSInAppMessage *)message {
             [self.messageDisplayQueue addObject:message];
         
         // Return early if an IAM is already showing
-        if (self.isInAppMessageShowing)
+        if (self.isInAppMessageShowing || !OneSignal.isRegisterUserFinished) {
```

Comment:
Maybe we can add a comment regarding this conditional addition. I imagine you are not putting it on  `[self evaluateMessages];` because even if the user is not registered, some triggers might have happened that make an IAM appear, but we need to wait for the user registration. Right? 

Can it happen that after the session call, the IAM added to the queue is deleted from the Server? Might messageDisplayQueue end in a weird consistency situation? because messageDisplayQueue is not used until evaluateMessageDisplayQueue is called

Reply:
Ya thats a good point I think I should check earlier so that the message doesn't enter the queue. I don't want the message in the queue and not in the messages array